src/teehr/metrics/gumboot_bootstrap.py

```diff
@@ -0,0 +1,188 @@
+"""Bootstrap classes."""
+from typing import Any, Callable, Union
+from collections.abc import Generator as PyGenerator
+from pathlib import Path
+import logging
+
+from arch.bootstrap import IIDBootstrap
+from arch.bootstrap.base import (
+    # _loo_jackknife, # leave one out jackknife
+    _add_extra_kwargs
+)
+from arch.typing import ArrayLike, Int64Array, Float64Array
+from numpy.random import Generator, RandomState
+import numpy as np
+import pandas as pd
+
+logger = logging.getLogger(__name__)
+
+# FILL_VALUE = -9999  # JAB
+
+
+class GumbootBootstrap(IIDBootstrap):
+    """Python implementation of the Gumboot R package."""
+
+    def __init__(
+        self,
+        *args: ArrayLike,
+        value_time: pd.Series,
+        water_year_month: Union[int, None] = None,
```

Comment:
Do some of these arguments have default values?  like `water_year_month`?  If it is optional, should the default be set here?